### PR TITLE
Restructure User Guide

### PR DESCRIPTION
**Pull request by mynameizzhafeez**
Key changes:

- Added table of contents
- Command explanation should follow a more rigid format.
- Explanations for commands should maximum include one image (the rest can just dump into examples)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -2,158 +2,474 @@
...
+      1. [**Exit the application `exit`**](#exit-the-application-exit)
+      2. [**Get help `help`**](#get-help-help)
+      3. [**List all contacts `list`**](#list-all-contacts-list)
+      4. [**Add a contact `add`**](#add-a-contact-add)
+      5. [**View a contact's profile `view`**](#view-a-contacts-profile-view)
+      6. [**Edit a contact's details `edit`**](#edit-a-contacts-details-edit)
+      7. [**Delete a contact `delete`**](#delete-a-contact-delete)
+      8. [**Add a label to a contact `tag`**](#tagging-a-module-to-an-existing-contact--tag)
+      9. [**Remove a label from a contact `tag`**](#untagging-a-module-from-an-existing-contact--untag)
+   4. [**Search Commands**](#search-commands)
```
Comment (zichen-3974) [inline docs/UserGuide.md:51]: Would it perhaps be better to group search, storage, meet under advanced commands? It feels slightly inconsistent to have basic, advanced, and 3 independent commands.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -2,158 +2,476 @@
...
+* `delete INDEX`: removes the contact at the specified `INDEX`.
+
+What you should see:<br>
+{GUI}<br>
+{Explanation}
+
+Examples:
+* `delete 4`: removes the fourth contact from the EduMate.
+
+### Tagging a module to an existing contact : `tag`
```
Comment (zichen-3974) [inline docs/UserGuide.md:474]: Should this be tag instead of tagging?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -235,98 +553,295 @@ Description of outcome:
 
 CS2103T is added to John Doe's, the user, list of modules.
 
-### Locating persons by keyowrds : `find`
+## Advanced Commands
+
+### Locating persons by keywords : `find`
```
Comment (zichen-3974) [inline docs/UserGuide.md:558]: Same for this, locate instead of locating?

File: docs/UserGuide.md (hunk 2 above)
Comment (mynameizzhafeez) [reply docs/UserGuide.md]: Alright, made changes.

**Pull request closed without merging**